### ISSUE TITLE
Software set idle cutoff not working in BMS

### ISSUE THREAD

**Issue by ajalberd** [chihirobelmo/FalconBMS-Alternative-Launcher#128]
I have a Virpil CM3 throttle and set the idle cutoff in launcher settings, enabling cutoff in BMS config as well. However, it doesn't apply in game, I cannot go in and out of idle. 

![image](https://github.com/chihirobelmo/FalconBMS-Alternative-Launcher/assets/20324292/141c1eb4-1722-4023-9144-1abc15940dce)

**Comment by arithex** [comment]
Can you clarify the version of AL (and BMS) you're running?  There was some bugs on this, that were fixed post-U3.

**Comment by ajalberd** [comment]
Everything's the latest version, BMS U4 as well.

**Comment by arithex** [comment]
I think we may still have a bug on dual-engine jets (F-15) but it appears working ok for me, on F-16.

If you're able to zip-up and share your Config folder, I'll take a deeper look.

**Comment by ajalberd** [comment]
[Config.zip](https://github.com/chihirobelmo/FalconBMS-Alternative-Launcher/files/15380109/Config.zip)

Thanks for looking into it @arithex . Here's my config.zip.

**Comment by arithex** [comment]
I see only `set g_bUseAnalogIdleCutoff 0` in falcon bms.cfg

You'll need to manually add a line, `set g_bUseAnalogIdleCutoff 1` in falcon bms user.cfg

(it's on my backlog to make AL smart enough to do that for you, if you set an idle-detent.. but it doesn't currently)

**Comment by ajalberd** [comment]
Yes, it is now, since enabling it makes both the cutoff **and** button not work. Right now I use button, sorry if I didn't mention

**Comment by arithex** [comment]
this is AL v2.4.1.8 (part of U4-beta)?

**Comment by ajalberd** [comment]
I'm trying 2.4.1.7 from here: https://github.com/BenchmarkSims/FalconBMS-Launcher/releases/tag/v2.4.1.7

There's a newer one?

**Comment by arithex** [comment]
should be fine .. not sure what to look at next.  is the problem with single-engine (F-16) or dual-engine (F-15) or both/different?

**Comment by ajalberd** [comment]
Seems to be both, thanks for your help anyways. It's strange, since I can set in launcher but it won't trigger in BMS actual, it shows up in BMS but won't work ingame.
![image](https://github.com/chihirobelmo/FalconBMS-Alternative-Launcher/assets/20324292/be803568-e20f-407a-851e-9976ab629fca)

**Comment by arithex** [comment]
does the CM3 have a "button" in the signalled state, when throttle is idle position?

I don't know, there may be some buggy interplay between the button-based callback signalling idle, and the analog-position based idle.   That hasn't been well tested..  the expectation is folks would do one or the other (analog or callback, not both).   I'll do some more testing..  your bug report spotlights a problem with dual-throttle still (thanks!)

**Comment by ajalberd** [comment]
The way I have it setup, no. Not sure the default, it's been many years. I'm using the alternate launcher to use the analog position based idle.

**Comment by ajalberd** [comment]
It actually seems to go out of idle now, but won't go back in. Meaning, when I choose ramp start, the throttle can be moved around its full range (even AB) but will never go back into idle. @arithex

**Comment by ajalberd** [comment]
Hmm, actually it works with a higher cutoff. Weird! But why doesn't the lower % one work?

I know BMS uses leading edge for throttle, where it doesn't register the input until I move it around some, could that be it?

![image](https://github.com/user-attachments/assets/b056670d-b0ad-4144-936d-8e069d2ec84a)

**Comment by arithex** [comment]
Oh dear, I must apologize -- my comments above were confused and unhelpful.  The reality is, g_bUseAnalogIdleCutoff simply isn't implemented for dual-throttle axes.  

I knew this, but somehow didn't remember clearly when you asked.  (The UX in Alt Launcher makes it look so plausible, I guess it fooled me.)

https://forum.falcon-bms.com/post/402590 for a little more info

Terribly sorry for those misleading responses, earlier.

**Comment by ajalberd** [comment]
No that's alright! My prior questions were for single-engine. I guess the problem was BMS doesn't register cutoff that low on the axis percent, moving it up works, even when it would be above my physical detent it doesn't go into idle until a bit below the red bar, so it works for me.

Shame on the F-15 part, though. Just as soon as I get it working for F-16 looks like it's useless for F15.

**Comment by arithex** [comment]
I'm still somewhat new to the team, and the codebase -- but yes, it was a disappointment to discover none of the code implementing g_bUseAnalogIdleCutoff references AXIS_THROTTLE2 at all.. and vice versa.   Will fix this, but it will take some time.

Btw there is some attempt in the code at enforcing a "minimum" or deadzone around the idle/cutoff detent position.. but it also appears buggy and inconsistent.  May explain what you're seeing, where it works better with a larger cutoff range.
